src/core/Database.cpp

```diff
@@ -202,8 +202,12 @@ bool Database::save(QString* error, bool atomic, bool backup)
  */
 bool Database::save(const QString& filePath, QString* error, bool atomic, bool backup)
 {
-    Q_ASSERT(!m_data.isReadOnly);
-    if (m_data.isReadOnly) {
+    // Disallow saving to the same file if read-only
+    if (m_data.isReadOnly && filePath == m_data.filePath) {
+        Q_ASSERT(false);
```

Comment:
I don't like this. You should at least use Q_ASSERT_X to give a future bug hunter some clue as to what went wrong.